types/pdfmake/interfaces.d.ts

```diff
@@ -0,0 +1,473 @@
+    | 'SRA3'
+    | 'SRA4'
+    | 'EXECUTIVE'
+    | 'FOLIO'
+    | 'LEGAL'
+    | 'LETTER'
+    | 'TABLOID';
+
+export type PageOrientation = 'portrait' | 'landscape';
+
+export type Size =
+    | number // absolute
+    | 'auto'
+    | '*'
+    | string; // percentage
+
+export interface TFontDictionnary {
+    [fontName: string]: TFontFamilyTypes;
+}
+
+export interface TFontFamilyTypes {
+    normal?: string;
+    bold?: string;
+    italics?: string;
+    bolditalics?: string;
+}
+
+export interface TDocumentInformation {
+    title?: string;
+    author?: string;
+    subject?: string;
+    keywords?: string;
+    creator?: string;
+    producer?: string;
+    creationDate?: Date;
+    modDate?: Date;
+    trapped?: string;
+}
+
+export type DynamicContent = (
+    currentPage: number,
+    pageCount: number,
+    pageSize: ContextPageSize,
+) => Content | null | undefined;
+
+export type DynamicBackground = (currentPage: number, pageSize: ContextPageSize) => Content | null | undefined;
+
+export type Margins = number | [number, number] | [number, number, number, number];
+
+export type Decoration = 'underline' | 'lineThrough' | 'overline';
+export type DecorationStyle = 'dashed' | 'dotted' | 'double' | 'wavy';
+
+export type Alignment = 'left' | 'right' | 'justify' | 'center';
+
+export type DynamicRowSize = (row: number) => number | 'auto';
+
+export interface CustomTableLayout {
+    hLineWidth?: DynamicLayout<number>;
+    vLineWidth?: DynamicLayout<number>;
+    hLineColor?: string | DynamicLayout<string>;
+    vLineColor?: string | DynamicLayout<string>;
+    hLineStyle?: DynamicLayout<LineStyle>;
+    vLineStyle?: DynamicLayout<LineStyle>;
+    fillColor?: string | DynamicLayout<string>;
+    paddingLeft?: number | DynamicLayout<number>;
+    paddingRight?: number | DynamicLayout<number>;
+    paddingTop?: DynamicLayout<number>;
+    paddingBottom?: DynamicLayout<number>;
+    fillOpacity?: number | DynamicLayout<number>;
+    defaultBorder?: boolean;
+}
+
+export type DynamicLayout<T> = (rowIndex: number, node: ContentTable, columnIndex: number) => T | null | undefined;
+
+export interface LineStyle {
+    dash?: {
+        length: number;
+        space?: number;
+    };
+}
+
+export type TableCell =
+    | {} // Used when another cell spans over this cell
+    | (Content & {
+          rowSpan?: number;
+          colSpan?: number;
+          border?: [boolean, boolean, boolean, boolean];
+          borderColor?: [string, string, string, string];
+          fillOpacity?: number;
+      });
+
+export interface Table {
+    body: TableCell[][];
+    widths?: Size[];
+    heights?: number | number[] | DynamicRowSize;
+    headerRows?: number;
+    dontBreakRows?: boolean;
+    keepWithHeaderRows?: number;
+    layout?: TableLayout;
+}
+
+export type PredefinedTableLayout = 'noBorders' | 'headerLineOnly' | 'lightHorizontalLines';
+export type TableLayout = PredefinedTableLayout | CustomTableLayout;
+
+export interface Style {
```

Comment:
Please add back type annotation comments.

Reply:
I didn't pay attention while merging, will do.

Reply:
This looks good.